package/MDAnalysis/analysis/dihedrals.py

```diff
@@ -0,0 +1,105 @@
+import numpy as np
+import matplotlib.pyplot as plt
+import warnings
+import six
+from six.moves import zip
+
+import MDAnalysis as mda
+from MDAnalysis.analysis.base import AnalysisBase
+from MDAnalysis.lib.distances import calc_dihedrals
+
+
+class Ramachandran(AnalysisBase):
+    """Calculate phi and psi dihedral angles of specified residues.
+
+    Note
+    ----
+    Run the analysis with :meth:`Ramachandran.run()`, which stores the results
+    in the array :attr:`Ramachandran.angles`. A axes object can be obtained
+    with :meth: `Ramachandran.run().plot()`.
+
+    If the residue selection is beyond the scope of the protein, then an error
+    will be raised. If the residue selection includes the first or last residue
+    then a warning will be raised, and the final array of angles will not
+    include those residues.
+
+    """
+    def __init__(self, atomgroup, **kwargs):
+        r"""Parameters
+        ----------
+        atomgroup : Atomgroup
+            atoms for residues for which phi and psi are calculated
+        start : int, optional
+            starting frame, default None becomes 0.
+        stop : int, optional
+            Frame index to stop analysis. Default: None becomes
+            n_frames. Iteration stops *before* this frame number,
+            which means that the trajectory would be read until the end.
+        step : int, optional
+            step between frames, default None becomes 1.
+
+        """
+        super(Ramachandran, self).__init__(atomgroup.universe.trajectory, **kwargs)
+        self.atomgroup = atomgroup
+        residues = self.atomgroup.residues
+        protein = self.atomgroup.universe.select_atoms("protein").residues
+
+        if not residues.issubset(protein):
+            raise IndexError("Found atoms outside of protein. Only atoms "
```

Comment:
I would use a value error here. The `IndexError` is when you use the square bracket operator.